package/kernel/mac80211/patches/ath9k/548-ath9k_enable_gpio_chip.patch

```diff
@@ -62,10 +61,9 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 +/* gpio_chip handler : set GPIO to input */
 +static int ath9k_gpio_pin_cfg_input(struct gpio_chip *chip, unsigned offset)
 +{
-+	struct ath9k_gpio_chip *gc = container_of(chip, struct ath9k_gpio_chip,
-+						  gchip);
++	struct ath_softc *sc = gpiochip_get_data(chip);
```

Comment:
I haven't tested it yet but it seems that the return value here is the address of the struct `ath9k_gpio_chip`.

Reply:
AFAIK it depends on the pointer passed to gpiochip_add_data. I'm passing in sc.

Reply:
btw do you know why owner is set to NULL?

Reply:
> AFAIK it depends on the pointer passed to gpiochip_add_data. I'm passing in sc.

Oh right, I see it.

> btw do you know why owner is set to NULL?

I think it does nothing now and can be removed. Before this [commit](https://github.com/torvalds/linux/commit/ff2b1359229927563addbf2f5ad480660c350903), the final gchip.owner will be override as NULL from gchip->parent->driver->owner. But now it's useless because it won't be automatically assigned anymore.